core/src/main/java/org/opensearch/sql/expression/HighlightExpression.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.expression;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.List;
+import lombok.Getter;
+import org.opensearch.sql.common.utils.StringUtils;
+import org.opensearch.sql.data.model.ExprTupleValue;
+import org.opensearch.sql.data.model.ExprValue;
+import org.opensearch.sql.data.model.ExprValueUtils;
+import org.opensearch.sql.data.type.ExprCoreType;
+import org.opensearch.sql.data.type.ExprType;
+import org.opensearch.sql.expression.env.Environment;
+import org.opensearch.sql.expression.function.BuiltinFunctionName;
+
+/**
+ * Highlight Expression.
+ */
+@Getter
+public class HighlightExpression extends FunctionExpression {
+  private final Expression highlightField;
+
+  /**
+   * HighlightExpression Constructor.
+   * @param highlightField : Highlight field for expression.
+   */
+  public HighlightExpression(Expression highlightField) {
+    super(BuiltinFunctionName.HIGHLIGHT.getName(), List.of(highlightField));
+    this.highlightField = highlightField;
+  }
+
+  /**
+   * Return String or Map value matching highlight field.
+   * @param valueEnv : Dataset to parse value from.
+   * @return : String or Map value of highlight fields.
+   */
+  @Override
+  public ExprValue valueOf(Environment<Expression, ExprValue> valueEnv) {
```

Comment:
As I understand, `OpenSearch` parses `SearchHits` and add `_highlight.xxx` to value env. Here we try to find `_highlight.xxx` and convert it to `highlight(xxx)` which is specified in `SELECT` by user. I'm thinking is it possible to spare the intermediate format (`_highlight.xxx`)? If so, is this new expression still required?

```
{
   "hits": [{
    "highlight": {
      "name": "..."
      "address": "..."
    }
  }]
}

=>

"_highlight.name"=...
"_highlight.address"=...

=>

highlight("name")
highlight("address")
```

Reply:
The intermediate format of `_highlight.xxx` is not overly important but we did need some sort of way to parse the highlight fields out of `valueEnv`. In [OpenSearchResponse](https://github.com/opensearch-project/sql/blob/6b96ac04e559160611cd9269fbf08dae02dd372a/opensearch/src/main/java/org/opensearch/sql/opensearch/response/OpenSearchResponse.java#L97-L111) revisions we must create a map of the highlight fields to use the expression and return the list of fields. I don't see a way that we could eliminate the expression entirely and still return single or multiple highlight fields.

Reply:
Okay. I was thinking in OpenSearchResponse can we return `highlight(field)` directly. Ignore this if impossible. Thanks.

Reply:
It just messes up if you pass `highlight("*")`.  We needed a workaround for that.